client/src/app.js

```diff
@@ -57,45 +91,28 @@ const warnLogout = _.debounce(() => {
   );
 }, 1000);
 
-// afterware for responses
-networkInterface.useAfter([
-  {
-    applyAfterware({ response }, next) {
-      if (!response.ok) {
-        response
-          .clone()
-          .text()
-          .then((bodyText) => {
-            console.log(`Network Error: ${response.status} (${response.statusText}) - ${bodyText}`);
-            next();
-          });
-      } else {
-        let isUnauthorized = false;
-        response
-          .clone()
-          .json()
-          .then(({ errors }) => {
-            if (errors) {
-              console.log('GraphQL Errors:', errors);
-              if (_.some(errors, e => e.message.startsWith('Unauthorized'))) {
-                isUnauthorized = true;
-              }
-            }
-          })
-          .then(() => {
-            if (isUnauthorized) {
-              warnLogout();
-              store.dispatch(logout());
-            }
-            next();
-          });
-      }
-    },
-  },
-]);
+// Hack to get around known issue with networkError not having network header errors
+// https://github.com/apollographql/apollo-link/issues/300
+const logoutLink = onError(({ graphQLErrors }) => {
+  if (graphQLErrors && graphQLErrors[0].message.includes('Unauthorized')) {
+    store.dispatch(logout());
+    return warnLogout();
+  }
+  return null;
+});
+
+// This should work once apollo-link issue #300 is resolved
```

Comment:
I really don't like leaving commented code in the repo - stash it in a branch or something. I won't reject the PR over this.